runtime/src/main/jni/ObjectManager.cpp

```diff
@@ -328,7 +328,17 @@ void ObjectManager::ReleaseJSInstance(Persistent<Object>* po, JSInstanceInfo* js
 
     m_idToObject.erase(it);
     m_released.insert(po, javaObjectID);
+
+    // In case the MarkReachableObjects is perfect, the regular instances will be reachable from an implementation object and won't be released.
+    // However, getting by mistake here, leads to referencing dangling pointers downstream, trying to get objects by weird ids.
+    // That's why we are clearing the external JSInstanceInfo* pointer from the JavaScript instance here.
```

Comment:
There were some old bug reports that apps ended up in "can't find object with id -213123" where the numbers appear pretty random. When I was working without MarkReachableObjects, I managed to reporduce one of these, the problem was that later after the deletion here, a method is called on a JavaScript instance that has its JsInfo deleted and probably its id was overwritten. Deleting the external reference here should allow us to throw better error messages for such JavaScript instances. If you still need the pointer to the deleted JsInfo.